packages/block-editor/src/components/block-title/use-block-display-title.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { truncate } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import {
+	getBlockType,
+	__experimentalGetBlockLabel as getBlockLabel,
+	isReusableBlock,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import useBlockDisplayInformation from '../use-block-display-information';
+import { store as blockEditorStore } from '../../store';
+
+export const MAXIMUM_TITLE_LENGTH = 35;
+
+/**
+ * Returns the block's configured title as a string, or empty if the title
+ * cannot be determined.
+ *
+ * @example
+ *
+ * ```js
+ * useBlockDisplayTitle( 'afd1cb17-2c08-4e7a-91be-007ba7ddc3a1', 17 );
+ * ```
+ *
+ * @param {string} clientId Client ID of block.
+ * @param {number} maximumLength The maximum length that the block title string may be before truncated.
+ * @return {?string} Block title.
+ */
+export default function useBlockDisplayTitle( clientId, maximumLength ) {
+	maximumLength = Number.isInteger( maximumLength )
+		? maximumLength
+		: MAXIMUM_TITLE_LENGTH;
```

Comment:
Could `maximumLength` be treated as an optional param with a default value?
```suggestion
 * @param {string} clientId Client ID of block.
 * @param {number|undefined} maximumLength The maximum length that the block title string may be before truncated.
 * @return {?string} Block title.
 */
export default function useBlockDisplayTitle( clientId, maximumLength = MAXIMUM_TITLE_LENGTH ) {
```

Reply:
Now that I've said that I've noticed that there doesn't seem to be a way to always get the full block title without any truncation. The only option is to use a really big number for the max length.

Because a new API is being introduced in this hook there's an opportunity to step away from the way `BlockTitle` did things.

The truncation in `BlockTitle` could still be left working as it was for back compat by passing in a max length param to the hook, but maybe when `undefined` is passed to the hook it should return the full block title (and not have an implicit default at all).

Reply:
Right, that was my initial thinking that isn't possible to show the full title with the revised implementation. There is some legacy handling here that truncates long titles for reusable blocks but it wasn't applied to other blocks. So maybe we could keep it as the default, and truncate everything where an explicit value is passed.

Reply:
Thanks for your help @talldan and @gziolo 

> maybe when undefined is passed to the hook it should return the full block title (and not have an implicit default at all
> So maybe we could keep it as the default, and truncate everything when an explicit value is passed.

 I agree that it makes a lot of sense if truncation is optional. 

I've taken this route in f625f28f8d6b65f6a0bc181874801eb5fc85c35c since it allows for greater flexibility. 👍 

For all existing usages I'm passing `35` (the current default) for now to maintain the status quo and to keep this PR manageable. 

If we want to revisit individual lengths, I'm happy to do it in a follow up PR. How does that sound?